tests/container_checker/test_container_checker.py

```diff
@@ -0,0 +1,299 @@
+    pytest.mark.disable_loganalyzer
+]
+
+CONTAINER_CHECK_INTERVAL_SECS = 1
+CONTAINER_STOP_THRESHOLD_SECS = 30
+CONTAINER_RESTART_THRESHOLD_SECS = 180
+
+
+@pytest.fixture(autouse=True, scope='module')
+def config_reload_after_tests(duthost):
+    yield
+    config_reload(duthost)
+
+
+@pytest.fixture(autouse=True, scope="module")
+def check_image_version(duthost):
+    """Skips this test if the SONiC image installed on DUT was 201911 or old version.
+
+    Args:
+        duthost: Host DUT.
+
+    Return:
+        None.
+    """
+    if parse_version(duthost.kernel_version) <= parse_version("4.9.0"):
+        pytest.skip("Test was not supported for 201911 and older image version!")
+
+
+def get_disabled_container_list(duthost):
+    """Gets the container/service names which are disabled.
+
+    Args:
+        duthost: Host DUT.
+
+    Return:
+        A list includes the names of disabled containers/services
+    """
+    disabled_containers = []
+
+    container_status, succeeded = duthost.get_feature_status()
+    pytest_assert(succeeded, "Failed to get status ('enabled'|'disabled') of containers. Exiting...")
+
+    for container_name, status in container_status.items():
+        if status == "disabled":
+            disabled_containers.append(container_name)
+
+    return disabled_containers
+
+
+def is_container_running(duthost, container_name):
+    """Decides whether the container is running or not
+
+    Args:
+        duthost: Host DUT.
+        container_name: Name of a container.
+    Return:
+        Boolean value. True represents the container is running
+    """
+    result = duthost.shell("docker inspect -f \{{\{{.State.Running\}}\}} {}".format(container_name))
+    return result["stdout_lines"][0].strip() == "true"
+
+
+def check_container_state(duthost, container_name, should_be_running):
+    """Determines whether a container is in the expected state (running/not running)
+
+    Args:
+        duthost: Host DUT.
+        container_name: Name of container.
+        should be running: Boolean value.
+
+    Return:
+        This function will return True if the container was in the expected state.
+        Otherwise, it will return False.
+    """
+    is_running = is_container_running(duthost, container_name)
+    return is_running == should_be_running
+
+
+def is_hiting_start_limit(duthost, container_name):
+    """Checks whether the container can not be restarted is due to start-limit-hit.
+
+    Args:
+        duthost: Host DUT.
+        container_name: name of a container.
+
+    Return:
+        If start limitation was hit, then this function will return True. Otherwise
+        it returns False.
+    """
+    service_status = duthost.shell("sudo systemctl status {}.service | grep 'Active'".format(container_name))
+    for line in service_status["stdout_lines"]:
+        if "start-limit-hit" in line:
+            return True
+
+    return False
+
+
+def clear_failed_flag_and_restart(duthost, container_name):
```

Comment:
Same definition is found in ```test_container_autorestart.py```. Could you help to make it a common utility in ```tests/common/helpers/dut_utils.py``` ? Thankss

Reply:
Great suggestion! Will move this function and its related function to the file `tests/common/helpers/dut_utils.py`.

Reply:
Updated!